core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,628 @@ def update_question_suggestion(
+        stats_models_to_update)
+
+
+def update_translation_contribution_stats(
+    suggestion, user_id, target_id, is_submission):
+    """Creates/updates TranslationContributionStatsModel model for
+    given translation submitter when a translation is submitted or reviewed.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            submitted or reviewed.
+        user_id: str. The ID of the corresponding submitter.
+        target_id: str. The ID of the target entity being suggested to.
+        is_submission: bool. A boolean value which indicates whether this
+            function is called when submitting a translation or not.
+    """
+    topic_id = ''
+    exp_opportunity_dict = (
+        opportunity_services.get_exploration_opportunity_summaries_by_ids(
+            [target_id]))
+    exp_opportunity = exp_opportunity_dict[target_id]
+    if exp_opportunity is not None:
+        topic_id = exp_opportunity.topic_id
+
+    content_plain_text = html_cleaner.strip_html_tags(suggestion.change.translation_html)
+    content_word_count = len(content_plain_text.split())
+
+    translation_contribution_stat_model = (
+        suggestion_models.TranslationContributionStatsModel.get(
+            suggestion.change.language_code, user_id, topic_id
+        ))
+
+    if translation_contribution_stat_model is None:
+        suggestion_models.TranslationContributionStatsModel.create(
+            language_code=suggestion.change.language_code,
+            contributor_user_id=user_id,
+            topic_id=topic_id,
+            submitted_translations_count=1,
+            submitted_translation_word_count=content_word_count,
+            accepted_translations_count=0,
+            accepted_translations_without_reviewer_edits_count=0,
+            accepted_translation_word_count=0,
+            rejected_translations_count=0,
+            rejected_translation_word_count=0,
+            contribution_dates=[
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date()
+            ]
+        )
+    else:
+        translation_contribution_stat = (
+            create_translation_contribution_stats_from_model(
+                translation_contribution_stat_model))
+
+        is_suggestion_accepted = (
+            suggestion.status == suggestion_models.STATUS_ACCEPTED)
+
+        is_suggestion_rejected = (
+            suggestion.status == suggestion_models.STATUS_REJECTED)
+
+        translation_contribution_stat.submitted_translations_count += 1 * int(is_submission)
+        translation_contribution_stat.submitted_translation_word_count += (
+            content_word_count) * int(is_submission)
+        translation_contribution_stat.accepted_translations_count += 1 * int(not is_submission and is_suggestion_accepted)
```

Comment:
I think you missed some line breaks here.

Why do you need the "1 *"?

Reply:
"1 *" is to add `accepted_translations_count` if the suggestion is not submission(i.e. when reviewing) and if the suggestion is accepted.